### PR TITLE
Fix #5490: Refactor TransactionView to improve performance

### LINKED ISSUE #5490: Wallet performance issue related to wei formatting
### Description: 

Some performance issues are being caught in Wallet when using the `WeiFormatter`:
image 

Causing poor scrolling performance:

https://user-images.githubusercontent.com/5314553/173389081-cae00f89-e51d-44d8-8fe8-ef1440abae45.MP4

We should try to cache formatter result to prevent using formatter on every SwiftUI layout pass.

### Steps to Reproduce 
If you have lots of transactions on one account
1. Open & unlock Wallet
2. Tap Accounts
3. Tap account with lots of transactions
4. Scroll the list of transactions up/down quickly, observe any micro-stutters:
    - My iPhone XS in above video shows the stutter

If you have lots of transactions on one token (can be across all accounts):
1. Open  & unlock Wallet
2. Tap token with lots of transactions in visible assets list
3. Scroll the list of transactions up/down quickly, observe any micro-stutters

### PR DESCRIPTION
**Pull request by StephenHeaps**
## Summary of Changes
- Extracted the displayed data for `TransactionView` into a `TransactionSummary` model. Previously we were using the `WeiFormatter` on every SwiftUI layout pass, this model allows us to generate the formatted values once and store in a model instead of on each layout pass.
- Created a `TransactionParser` to help generate this `TransactionSummary` model given a transaction and necessary values (the current network, user accounts, user visible assets, all tokens, and asset price ratios).
- Replaced `TransactionView` with a `TransactionSummaryView` and updated `AccountActivityStore` / `AccountActivityView` and `AssetDetailStore` / `AssetDetailView` to use this new `TransactionSummary` model and `TransactionSummaryView`.
- Added the new `ethSwap` so we display titles for swap transactions again.

This pull request fixes #5490

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:
If you have lots of transactions on one account
1. Open & unlock Wallet
2. Tap Accounts
3. Tap account with lots of transactions
4. Scroll the list of transactions up/down quickly, should now see smooth scrolling without micro-stutters

If you have lots of transactions on one token (can be across all accounts):
1. Open  & unlock Wallet
2. Tap token with lots of transactions in visible assets list
3. Scroll the list of transactions up/down quickly, should now see smooth scrolling without micro-stutters

Developers: Although most of the logic was copied when moving to the `TransactionSummary` / `TransactionParser` builder, I screenshotted both views against `development` and this branch to ensure the transactions display the same. Currently on `development` you will have to add `.ethSwap` to the case statements [here (Transactions/TransactionView.swift#L95)](https://github.com/brave/brave-ios/blob/development/BraveWallet/Crypto/Transactions/TransactionView.swift#L95) and [here (AssetDetailStore.swift#L185)](https://github.com/brave/brave-ios/blob/development/BraveWallet/Crypto/Stores/AssetDetailStore.swift#L185) to match up (otherwise swaps will be excluded as we don't correctly handle the new TxType).

## Screenshots:

Updated video from my XS (original stuttering video in Issue #5490):

https://user-images.githubusercontent.com/5314553/173398547-cee10b3c-7593-4a2a-968c-16277990a35f.MP4

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: BraveWallet/Crypto/Asset Details/AssetDetailView.swift [hunk 1]
```diff
@@ -77,28 +77,19 @@ struct AssetDetailView: View {
...
+            Button(action: { self.transactionDetails = txSummary.txInfo }) {
+              TransactionSummaryView(summary: txSummary, displayAccountCreator: true)
             }
             .contextMenu {
-              if !tx.txHash.isEmpty {
+              if !txSummary.txHash.isEmpty {
                 Button(action: {
                   if let baseURL = self.networkStore.selectedChain.blockExplorerUrls.first.map(URL.init(string:)),
-                    let url = baseURL?.appendingPathComponent("tx/\(tx.txHash)") {
+                     let url = baseURL?.appendingPathComponent("tx/\(txSummary.txHash)") {
```
Comment (nuo-xu) [inline BraveWallet/Crypto/Asset Details/AssetDetailView.swift:92]: seems like introduced one more space here

Comment (StephenHeaps) [reply BraveWallet/Crypto/Asset Details/AssetDetailView.swift:92]: Noticed this too, same in `AccountActivityView`, but re-indent remains the same. I believe we were missing a space before.

Comment (kylehickinson) [reply BraveWallet/Crypto/Asset Details/AssetDetailView.swift:92]: When we ran `swiftformat` back in the day it messed up this indentation, this is the correct Xcode indentation

File: BraveWallet/Crypto/Stores/AccountActivityStore.swift [hunk 2]
```diff
@@ -9,7 +9,7 @@ import BraveCore
 class AccountActivityStore: ObservableObject {
   private(set) var account: BraveWallet.AccountInfo
   @Published private(set) var assets: [AssetViewModel] = []
-  @Published private(set) var transactions: [BraveWallet.TransactionInfo] = []
+  @Published var transactionSummarys: [TransactionSummary] = []
```
Comment (kylehickinson) [inline BraveWallet/Crypto/Stores/AccountActivityStore.swift]: `transactionSummaries`

**Review by nuo-xu — APPROVED** [review]

File: BraveWallet/Crypto/Stores/AccountActivityStore.swift (hunk 2 above)
Comment (StephenHeaps) [reply BraveWallet/Crypto/Stores/AccountActivityStore.swift]: Fixed in [b740445](https://github.com/brave/brave-ios/pull/5512/commits/b740445d5a902ccbb5d4adf98f3a43ae818b5132)

**Review by kylehickinson — APPROVED** [review]

**Pull request merged**